src/reference/asciidoc/sftp.adoc

```diff
@@ -573,7 +573,12 @@ If you don't actually want to persist the state, an in-memory `SimpleMetadataSto
 If you wish to use a filename pattern (or regex) as well, use a `CompositeFileListFilter`.
 The java configuration below shows one technique to remove the remote file after processing.
 
-Use the `max-fetch-size` attribute to limit the number of files fetched on each poll when a fetch is necessary; set to 1 and use a persistent filter when running in a clustered environment;
+Use the `max-fetch-size` attribute to limit the number of files fetched on each poll when a fetch is necessary; set to 1 and use a persistent filter when running in a clustered environment.
+
+The adapter puts the remote directory and file name in headers `FileHeaders.REMOTE_DIRECTORY` and `FileHeaders.REMOTE_FILE` respectively.
+Starting with _version 5.0_, additional remote file information, in JSON, is provided in the `FileHeaders.REMOTE_FILE_INFO` header.
+If you set the `fileInfoJson` property on the `SftpStreamingMessageSource` to `false`, the `LsEntry` object provided by the underlying JSch library is used instead.
```

Comment:
Well, I see you document your solution.
But why this way?
Why that `FileInfo.getFileInfo()` isn't enough to get access to the underlying entity?

Seems for me our `FileInfo` is more convenient in most cases, especially when we have a common process after FTP as well as SFTP.
And as I said above: it isn't consistent with JSON representation.

Any additional argument to go your way?
Thanks